### PR TITLE
Feat: Edit Guardian Profile

### LINKED ISSUE #118: Feat: Edit Guardian Profile
Update the family information page to allow for editing the guardian information, make sure to include the ability to upload a profile picture

Image

### PR DESCRIPTION
**Pull request by gregorysa**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/118)
- Page for editing guardian profile
- Mutation function constant in auth context for updating guardian
- Support changing name, username, and profile pic
- Components for image selector and icon for empty profile pic
- Stored profile pic is used on profile and family information pages
- "X" button on edit page to remove profile pic
- Changing the language updates the language preference stored in the database
- useGuardian hook takes in a guardian ID

# How Has This Been Tested?
- Logged the currently stored language preference to check that it updated whenever the language is changed
- Updated guardian information is reflected in the database and persists after logging out and logging back in
- Changes to name, username, and profile pic are not saved until the submit button is pressed
- Profile pic can be selected and removed multiple times while editing
- "X" button only appears when there is an image selected or the user currently has a profile pic
- Profile pic stays after switching between languages

## Screenshots
Screenshot 2026-04-01 at 1 33 37 PM
Screenshot 2026-04-01 at 1 33 42 PM
Screenshot 2026-04-01 at 1 33 49 PM
Screenshot 2026-04-01 at 1 34 13 PM
Screenshot 2026-04-01 at 1 34 58 PM
Screenshot 2026-04-01 at 1 35 01 PM
Screenshot 2026-04-01 at 1 35 04 PM

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS

https://github.com/user-attachments/assets/8ed423bc-621f-440d-a310-6017656d5204

https://github.com/user-attachments/assets/45b7ea21-b687-4a1a-9d42-eed00a4903bc

https://github.com/user-attachments/assets/1d0c4089-48dd-48ad-ae54-baed22b0d4e6

### Android
[Add Android screenshots here]

### CODE REVIEW

**Review by adescoteaux1 — COMMENTED** [review]
at a glance, looks pretty good to me

File: frontend/apps/mobile/app/(app)/(tabs)/profile.tsx [hunk 1]
```diff
@@ -1,134 +1,144 @@
...
-        </View>
-        <View className="h-5" />
-      </ScrollView>
-    </ThemedView>
-  );
+	return (
+		<ThemedView className="flex-1" style={{ paddingTop: insets.top }}>
+			<ScrollView
+				showsVerticalScrollIndicator={false}
+				contentContainerStyle={{ paddingTop: 10, paddingBottom: 20 }}
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/profile.tsx]: Please use the className property for static styling

**Review by adescoteaux1 — APPROVED** [review]
